### PR TITLE
Albn/enhancement kspace

### PR DESCRIPTION
**Pull request by Ablinne**
Enhances the dumpreader by the capability to use the information of multiple fields in order to reconstruct the full k-space of polarizations

### CODE REVIEW

File: postpic/_field_calc.py [hunk 1]
```diff
@@ -242,3 +242,90 @@ def divE(self, **kwargs):
...
+        # the polaxis component of polfield
+        fields.append(field_factories[polfield][polaxis]())
+
+        # and the other two components of the otherfield needed for the cross-product
+        fields.append(field_factories[otherfield][(polaxis+1) % 3]())
+        fields.append(field_factories[otherfield][(polaxis+2) % 3]())
+
+        # find the origin of the simulation box, needed to unstagger the fields
+        dims = self.simdimensions()
+        simorigin = [self.simextent(i)[0] for i in range(dims)]
```
Comment (skuschel) [inline postpic/_field_calc.py]: E and B fields may cover only a part of the simulation box. Therefore the simorigin is quite meaningless. Probably better is to use the extent of the first field given.

Comment (Ablinne) [reply postpic/_field_calc.py]: Changed to rely on first given field

**Comment by Ablinne** [comment]
Remaining issues:

- If given fields have unequal number of grid points, function will fail
- If given fields have unequal box size but equal number of grid points, function will not fail, but results are nonsense
- If given fields have equal box size and equal number of grid points, but are different regions of simulation, results might also be nonsensical (except when the shift is very small as due to the field stagger)

**Comment by Ablinne** [comment]
> Remaining issues:
> 
>     If given fields have unequal number of grid points, function will fail
>     If given fields have unequal box size but equal number of grid points, function will not fail, but results are nonsense
>     If given fields have equal box size and equal number of grid points, but are different regions of simulation, results might also be nonsensical (except when the shift is very small as due to the field stagger)

These issues have been solved by checking the conditions and raise ValueError if the conditions are not met.

We could add convenience functions that make sure these conditions are fulfilled when cutting or slicing multiple Fields, but this must not necessarily happen within this pull request.

Anything else that must be considered before this can be merged?

**Pull request merged**